README.md

```diff
@@ -1 +1,204 @@
-# rust-vmm-ci
\ No newline at end of file
+# rust-vmm-ci
+
+The `rust-vmm-ci` contains [integration tests](#integration-tests) and
+[Buildkite pipeline](#buildkite-pipeline) definitions that are used for
+running the CI for all rust-vmm crates.
+
+Having a centralized place for the tests is one of the enablers for keeping the
+same quality standard for all crates in rust-vmm.
+
+## Getting Started with rust-vmm-ci
+
+To run the integration tests defined in the pipeline as part of the CI:
+
+1. Add rust-vmm-ci as a git submodule to your repository
+
+```bash
+git submodule add https://github/rust-vmm/rust-vmm-ci
+git add rust-vmm-ci/
+git commit -s -m "Added rust-vmm-ci as submodule"
+```
+
+2. Create the coverage test configuration file named coverage_config.json in
+the root of the repository. One example of the configuration file can be found
+in [coverage_config.json.sample](coverage_config.json.sample).
+
+The json must have the following fields:
+- *coverage_score*: The coverage of the repository.
+- *exclude_path*: This field is used for excluding files from the report. It
+  should be used to exclude autogenerated files. Files in *exclude_path* are
+  separated by one coma. If the repository does not have any autogenerated
+  files, *exclude_path* should be an empty string.
+- *crate_features*: `cargo kcov` does not build crate features by default. To
+  get the coverage report including optional features these need to be
+  specified in *crate_features* separated by coma. If the crate does not have
+  any features, this field should be empty.
+
+This file is required for the coverage integration so it needs to be added
+to the repository as well.
+
+3. Create a new pipeline definition in Buildkite. For this step ask one of the
+rust-vmm Buildkite [admins](CODEOWNERS) to create one for you. Add a pipeline
+step that is uploading the rust-vmm-ci pipeline:
+
+```bash
+buildkite-agent pipeline upload rust-vmm-ci/.buildkite/pipeline.yml
+```
+
+4. The code owner of the repository will have to setup a WebHook for triggering
+the CI on Pull Requests and pushes.
+
+## Buildkite Pipeline
+
+The [Buildkite](https://buildkite.com) pipeline is the definition of tests to
+be run as part of the CI. It includes steps for running unit tests and linters
+(including coding style checks), and computing the coverage.
+
+Currently the tests run on a Linux machine on x86_64 and aarch64.
```

Comment:
Nit: `a` here feels like it's running on a specific machine. I'd rephrase it as:
`Currently the tests can run on Linux x86_64 and aarch64 hosts`